<launch>

<arg name="robot_namespace"       default="$(optenv DINGO_CUSTOM_NAME d)"/>
<!-- Generate the tf_prefix from the namespace -->
<arg name="tf_prefix" default="$(eval optenv('TF_PREFIX', arg('robot_namespace') + '_tf_'))" />

<group ns="$(optenv DINGO_CUSTOM_NAME d)">
	<node name="collision_avoidance_2d" pkg="oarbot_control" type="collision_avoidance_2d.py" output="screen">
		<rosparam  command="load" file="$(find assistive_launch)/config/dingo_collision_avoidance_2d.yaml"/>

		<rosparam param="tf_world_frame_id"         subst_value="true">map</rosparam>
		<rosparam param="tf_mobile_base_frame_id"   subst_value="true">$(arg tf_prefix)base_link</rosparam>
	</node>
</group>
</launch>